Change of mind in asking for appointment.

29-29 opened the pull request:
As Student, 
I want to See a cancel button, 
So that Just in case I changed my mind to cancel book for an appointment.

29-29 committed:
added cancel booking function to the cancel action

29-29 committed:
implemented booking cancellation

29-29 committed:
fixed sonar issues

lucman-abdulrachman commented on src/pages/booking/EditBookingPage.vue:
```diff
@@ -219,7 +219,7 @@ const isPartOfIt = computed(() => {
   return false;
 });
-const busy = ref(false);
+const busy = ref(false); // what is this for?
```
you set it busy so that you can reference in :loading prop of the button preventing user from re-clicking submit while the system is processing at backend. 
YOu may remove the comment.

lucman-abdulrachman commented on src/workflows/booking/cancel-booking.workflow.ts:
```diff
@@ -0,0 +1,22 @@
+import { TheWorkflows } from '../the-workflows';
+
+TheWorkflows.on({
+  type: 'cancelBooking',
+  async cb(e) {
```
```suggestion
  type: 'cancelBooking',
  loggable: 'operation',
  async cb(e) {
```
this will allow viewing operation history of the booking later.

lucman-abdulrachman reviewed (dismissed):
see comments

lucman-abdulrachman committed:
Merge branch 'develop' into ES-3

29-29 committed:
resolved suggested changes

- removed comment on busy ref
- added loggable property to cancelbooking workflow

29-29 committed:
Merge branch 'ES-3' of https://github.com/MSU-Students/student-advising-portal into ES-3
